### DISCUSSION TITLE
searching for non-code items

### DISCUSSION THREAD

**Discussion by ChardeliusMe** [community/community#129080] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Hi! I'm currently working on a project that requires a ton of Github files. This including non-code files. But if i search in a project for a part of the name, it will at first only show me 40 items. Then, if i press Show more, it will search in CODE. But i don't want to search in code. so PLEASE, github, make a possibility to search for non-code items. Now i will have to manually look through about 200 folders.

**Comment by asdacosta** [comment]
Refine searches with file type filters (e.g., `filename:*.md`)

**Comment by look** [comment]
Can you clarify what you would like to do? It sounds like you want to search by file path rather than the file contents. You can use code search for that with the `path:` qualifier.

**Reply by XDYB** [reply]
When will non-UTF8 searches be supported?

**Reply by look** [reply]
It is not on our roadmap.

**Comment by taylortn** [comment]
Response #2

I understand the challenge you're facing with searching for non-code files in a large GitHub project. While GitHub's built-in search functionality is primarily focused on code, there are a few strategies and tools you can use to effectively search for non-code files. Here are some suggestions:

### Using GitHub's Advanced Search

Although GitHub's search is primarily for code, you can use some advanced search operators to filter results more effectively. For example:

1. **Filename Search**: Use the `filename` filter to search for files by name.
   ```plaintext
   filename:part_of_filename
   ```

2. **Path Search**: If you know the directory, you can search within a specific path.
   ```plaintext
   path:/path/to/directory/ filename:part_of_filename
   ```

### Using a Local Clone and Command-Line Tools

Cloning the repository locally and using command-line tools can be a powerful way to search for non-code files.

1. **Clone the Repository**:
   ```bash
   git clone https://github.com/your-repo-url.git
   cd your-repo-name
   ```

2. **Use `find` Command**: Use the `find` command to search for files by name.
   ```bash
   find . -type f -name "*part_of_filename*"
   ```

3. **Use `grep` Command**: If you need to search for content within non-code files.
   ```bash
   grep -r "search_term" .
   ```

### Using Third-Party Tools

There are third-party tools and extensions that can enhance your search capabilities:

1. **GitHub Desktop**: This desktop application provides a more visual interface for managing repositories and may make it easier to browse and search files.
   - [GitHub Desktop](https://desktop.github.com/)

2. **Visual Studio Code**: VS Code, with its extensions, can provide advanced search capabilities. You can use the built-in search feature to find files and content within a project.
   - [Visual Studio Code](https://code.visualstudio.com/)

3. **Sourcegraph**: Sourcegraph provides powerful search capabilities for large codebases, including non-code files.
   - [Sourcegraph](https://sourcegraph.com/)

### Custom Scripts

If you have specific requirements, writing a custom script to search for files might be the best solution. Here's a simple Python script to search for files by name:

```python
import os

def search_files(root_dir, search_term):
    matches = []
    for root, dirnames, filenames in os.walk(root_dir):
        for filename in filenames:
            if search_term in filename:
                matches.append(os.path.join(root, filename))
    return matches

# Example usage
root_directory = 'path_to_your_local_repo'
search_term = 'part_of_filename'
matches = search_files(root_directory, search_term)

for the match in matches:
    print(match)
```

### Summary

While GitHub's native search may have limitations, using a combination of advanced search operators, local command-line tools, third-party applications, and custom scripts can help you efficiently search for non-code files in large projects. If you frequently encounter this issue, consider providing feedback to GitHub to enhance their search capabilities for non-code files.
